File: ios/Classes/InstagramsharePlugin.m
#import "InstagramsharePlugin.h"
#import <instagram_share/instagram_share-Swift.h>

@implementation InstagramsharePlugin
+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
  [SwiftInstagramsharePlugin registerWithRegistrar:registrar];
}
@end

File: ios/Classes/InstagramSharePlugin.swift
import Flutter
import UIKit

public class InstagramSharePlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(name: "instagram_share", binaryMessenger: registrar.messenger())
    let instance = InstagramSharePlugin()
    registrar.addMethodCallDelegate(instance, channel: channel)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    result("iOS " + UIDevice.current.systemVersion)
  }
}
